src/Microsoft.Diagnostics.DebugServices.Implementation/SymbolService.cs

```diff
@@ -0,0 +1,249 @@
+        {
+            SymbolReader.DisableSymbolStore();
+        }
+
+        /// <summary>
+        /// Download a file from the symbol stores/server.
+        /// </summary>
+        /// <param name="key">index of the file to download</param>
+        /// <returns>path to the downloaded file either in the cache or in the temp directory or null if error</returns>
+        public string DownloadFile(SymbolStoreKey key)
+        {
+            return SymbolReader.GetSymbolFile(key);
+        }
+
+        /// <summary>
+        /// Attempts to download/retrieve from cache the key.
+        /// </summary>
+        /// <param name="key">index of the file to retrieve</param>
+        /// <returns>stream or null</returns>
+        public SymbolStoreFile GetSymbolStoreFile(SymbolStoreKey key)
+        {
+            if (IsSymbolStoreEnabled)
+            {
+                return SymbolReader.GetSymbolStoreFile(key);
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Returns the metadata for the assembly
+        /// </summary>
+        /// <param name="imagePath">file name and path to module</param>
+        /// <param name="imageTimestamp">module timestamp</param>
+        /// <param name="imageSize">size of PE image</param>
+        /// <returns>metadata</returns>
+        public ImmutableArray<byte> GetMetadata(string imagePath, uint imageTimestamp, uint imageSize)
+        {
+            try
+            {
+                Stream peStream = null;
+                if (imagePath != null && File.Exists(imagePath))
+                {
+                    peStream = TryOpenFile(imagePath);
+                }
+                else if (IsSymbolStoreEnabled)
+                {
+                    SymbolStoreKey key = PEFileKeyGenerator.GetKey(imagePath, imageTimestamp, imageSize);
+                    peStream = GetSymbolStoreFile(key)?.Stream;
+                }
+                if (peStream != null)
+                {
+                    using var peReader = new PEReader(peStream, PEStreamOptions.Default);
+                    if (peReader.HasMetadata)
+                    {
+                        PEMemoryBlock metadataInfo = peReader.GetMetadata();
+                        return metadataInfo.GetContent();
+                    }
+                }
+            }
+            catch (Exception ex) when 
+                (ex is UnauthorizedAccessException || 
+                 ex is BadImageFormatException || 
+                 ex is InvalidVirtualAddressException || 
+                 ex is IOException)
+            {
+                Trace.TraceError($"GetMetaData: {ex.Message}");
+            }
+            return ImmutableArray<byte>.Empty;
+        }
+
+        /// <summary>
+        /// Displays the symbol server and cache configuration
+        /// </summary>
+        public override string ToString()
+        {
+            StringBuilder sb = new StringBuilder();
```

Comment:
`return string.Empty`? Or was the goal here different?

Reply:
This will be replaced in Part 3. The whole SymbolService implementation will be different (SOS.NETCore is going away)